Rock/Model/CodeGenerated/DataViewService.cs

```diff
@@ -83,6 +84,55 @@ public bool CanDelete( DataView item, out string errorMessage )
             }  
             return true;
         }
+
+        /// <summary>
+        /// Create a new non-persisted Data View using an existing Data View as a template. 
+        /// </summary>
+        /// <param name="dataViewId">The identifier of a Data View to use as a template for the new Data View.</param>
+        /// <returns></returns>
+        public DataView GetNewFromTemplate(int dataViewId)
+        {
+            var item = this.Queryable()
+                           .AsNoTracking()
+                           .Include( x => x.DataViewFilter )
+                           .FirstOrDefault( x => x.Id == dataViewId );
+
+            if (item == null)
```

Comment:
@mrupsidedown I wrote about that precise topic here as a reminder to myself
http://blog.arranfrance.com/arrans-guide-to-fixing-a-pull-request/
On 13 Aug 2015 22:40, "MrUpsideDown" <notifications@github.com> wrote:

> In Rock/Model/CodeGenerated/DataViewService.cs
> <https://github.com/SparkDevNetwork/Rock/pull/1187#discussion_r37029001>:
>
> > @@ -83,6 +84,55 @@ public bool CanDelete( DataView item, out string errorMessage )
> >              }
> >              return true;
> >          }
> > +
> > +        /// <summary>
> > +        /// Create a new non-persisted Data View using an existing Data View as a template.
> > +        /// </summary>
> > +        /// <param name="dataViewId">The identifier of a Data View to use as a template for the new Data View.</param>
> > +        /// <returns></returns>
> > +        public DataView GetNewFromTemplate(int dataViewId)
> > +        {
> > +            var item = this.Queryable()
> > +                           .AsNoTracking()
> > +                           .Include( x => x.DataViewFilter )
> > +                           .FirstOrDefault( x => x.Id == dataViewId );
> > +
> > +            if (item == null)
>
> @dcs619 <https://github.com/dcs619> Perhaps you can help me here. On the
> topic of minor amendments to PRs, what is the best method for submitting
> changes like this? In the past, I have tried simply pushing a new commit to
> the branch in my forked repo, but this doesn't squash the commits in the
> PR, just adds a second commit (which is not preferred as I understand it).
> Is there any way to do this easily, because otherwise it requires me to
> close and re-create the PR (which is time-consuming)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SparkDevNetwork/Rock/pull/1187/files#r37029001>.
>